src/algebra/algebra/subalgebra.lean

```diff
@@ -258,6 +258,25 @@ instance integral_domain {R A : Type*} [comm_ring R] [integral_domain A] [algebr
   (S : subalgebra R A) : integral_domain S :=
 @subring.domain A _ S _
 
+/-- The equalizer of two R-algebra homomorphisms -/
+def equalizer (ϕ ψ : A →ₐ[R] B) : subalgebra R A :=
+{ carrier := {a | ϕ a = ψ a},
+  zero_mem' := by { change ϕ 0 = ψ 0, rw [alg_hom.map_zero, alg_hom.map_zero] },
+  add_mem' := λ x y hx hy, by
+  { change ϕ x = ψ x at hx,
+    change ϕ y = ψ y at hy,
+    change ϕ (x + y) = ψ (x + y),
+    rw [alg_hom.map_add, alg_hom.map_add, hx, hy] },
+  one_mem' := by { change ϕ 1 = ψ 1, rw [alg_hom.map_one, alg_hom.map_one] },
+  mul_mem' := λ x y hx hy, by
+  { change ϕ x = ψ x at hx,
+    change ϕ y = ψ y at hy,
+    change ϕ (x * y) = ψ (x * y),
+    rw [alg_hom.map_mul, alg_hom.map_mul, hx, hy] },
+  algebra_map_mem' := λ x, by
+  { change ϕ (algebra_map R A x) = ψ (algebra_map R A x),
+    rw [alg_hom.commutes, alg_hom.commutes] } }
+
```

Comment:
It would be good to add a `simp`-lemma `mem_equalizer`.
Also, in which namespace is this definition? If we put it in `alg_hom`, then we can write `f.equalizer g` when `f` and `g` are two alg_homs.

Reply:
I moved it into an `alg_hom` namespace and added `mem_equalizer`